gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/taskexecution/violations/GetProject.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone.taskexecution.violations;
+
+import com.google.errorprone.VisitorState;
+import com.palantir.gradle.guide.errorprone.taskexecution.IllegalMethodCalledDuringTaskExecution;
+import com.palantir.gradle.guide.errorprone.utils.Tasks;
+import com.sun.source.tree.MethodInvocationTree;
+
+public final class GetProject implements Violation {
+    private final IllegalMethodCalledDuringTaskExecution illegalMethodCalledDuringTaskExecution;
+
+    public GetProject(IllegalMethodCalledDuringTaskExecution illegalMethodCalledDuringTaskExecution) {
```

Comment:
Rather than injecting `IllegalMethodCalledDuringTaskExecution` into every `Violation` class, can we just give a `Description` in `fixOrReport` that has already been created using `state.reportMatch(illegalMethodCalledDuringTaskExecution.buildDescription(illegalCall)`. I think we could avoid needing the `tree` or the `state` in `fixOrReport` then and just have it return a `Description`.

Reply:
> I think we could avoid needing the tree or the state in fixOrReport then and just have it return a Description

In general I don't think it's possible to achieve a purely functional interface here. There's no running away from passing the `VisitorState` class, as it is needed for various AST Helper functions. e.g. to render a fix, we need `state.getSourceForNode`.